File: src/main/kotlin/com/leetcode/P1828.kt
package com.leetcode

import kotlin.math.sqrt

// https://github.com/antop-dev/algorithm/issues/565
class P1828 {
    fun countPoints(points: Array<IntArray>, queries: Array<IntArray>): IntArray {
        return IntArray(queries.size) { i ->
            val (px, py, r) = queries[i]
            count(px, py, r, points)
        }
    }

    private fun count(qx: Int, qy: Int, r: Int, points: Array<IntArray>): Int {
        return points.count { (px, py) ->
            px in qx - r..qx + r
                    && py in qy - r..qy + r
                    && distance(qx, qy, px, py) <= r
        }
    }

    // 두 점의 거리 구하기
    private fun distance(qx: Int, qy: Int, px: Int, py: Int): Double {
        val xDist = px.toDouble() - qx
        val yDist = py.toDouble() - qy
        return sqrt(xDist * xDist + yDist * yDist)
    }

}

File: src/python3/com/leetcode/p1828.py
import math
from typing import List


class Solution:
    def countPoints(self, points: List[List[int]], queries: List[List[int]]) -> List[int]:
        ans = [0] * len(queries)
        for i in range(len(queries)):
            px, py, r = queries[i]
            ans[i] = self.count(px, py, r, points)
        return ans

    def count(self, qx, qy, r, points):
        count = 0
        for px, py in points:
            if qx - r <= px <= qx + r \
                    and qy - r <= py <= qy + r \
                    and self.distance(qx, qy, px, py) <= r:
                count += 1
        return count

    def distance(self, qx, qy, px, py):
        x_dist = px - qx
        y_dist = py - qy
        return math.sqrt(x_dist ** 2 + y_dist ** 2)
